src/android/CameraLauncher.java

```diff
@@ -163,7 +159,19 @@ else if ((srcType == PHOTOLIBRARY) || (srcType == SAVEDPHOTOALBUM)) {
             PluginResult r = new PluginResult(PluginResult.Status.NO_RESULT);
             r.setKeepCallback(true);
             callbackContext.sendPluginResult(r);
-            
+
+            return true;
+        } else if (action.equals("checkForSavedResult")) {
+            if (savedRequestCode > 0 || savedResultCode > 0) {
+                processConfiguration(args);
+                imageUri = Uri.fromFile(createCaptureFile(encodingType));
+                onActivityResult(savedRequestCode, savedResultCode, savedIntent);
+                savedRequestCode = 0;
+                savedResultCode = 0;
+                savedIntent = null;
+            } else {
+                callbackContext.success("");
```

Comment:
one should only return an error when an error occurs.
@agrieve advised me to callback success with empty string so you can handle the result in onSuccess javascript like so:
`if (result) {
  //do something
}`